skimage/feature/util.py

```diff
@@ -145,16 +146,29 @@ def plot_matched_features(
     ax.imshow(image, cmap='gray')
     ax.axis((0, image0.shape[1] + offset[1], image0.shape[0] + offset[0], 0))
 
-    rng = np.random.default_rng()
+    rng = np.random.default_rng(seed=0)
 
-    for i in range(matches.shape[0]):
-        idx0 = matches[i, 0]
-        idx1 = matches[i, 1]
+    number_of_matches = matches.shape[0]
 
-        if matches_color is None:
-            color = rng.random(3)
-        else:
-            color = matches_color
+    from matplotlib.colors import is_color_like
+
+    if matches_color is None:
+        colors = [rng.random(3) for _ in range(number_of_matches)]
+    elif is_color_like(matches_color):
+        colors = [matches_color for _ in range(number_of_matches)]
+    elif isinstance(matches_color, Sequence):
```

Comment:
Should we apply `is_color_like` to this case, a sequence of colors is given, as well? Should lead to clearer error messages?

Reply:
I had the same thought but the error message you get from matplotlib itself is already better than what I would have come up with :grin:

> ValueError: 'x' is not a valid value for color: supported inputs are (r, g, b) and (r, g, b, a) 0-1 float tuples; '#rrggbb', '#rrggbbaa', '#rgb', '#rgba' strings; named color strings; string reprs of 0-1 floats for grayscale values; 'C0', 'C1', ... strings for colors of the color cycle; and pairs combining one of the above with an alpha value

Reply:
Ah, good to know. I added a test for that in https://github.com/scikit-image/scikit-image/pull/7541/commits/9abe2377875156e32886103cfc737b53c17f27c2.